client/collect.py

```diff
@@ -142,7 +143,11 @@ def run_dir(dir_path, parse_output=True, delete_after_success=False,
         else:
             run_file(*args, **kwargs)
     for thread in threads:
-        thread.join()
+        # Don't wait more than 1 day. 
+        relativeTimeout = timeout - time.time()
```

Comment:
I'm not sure why you're computing a timeout based on the current time, only to subtract the current time later. Why not just compute the relative timeout without current time? It looks like you're trying to impose a total time limit on the overall `run_dir` invocation, but I'm not sure why you'd want to do that. I'd remove the conditional breaks.

```python
thread_timeout=(60 * 60) # one hour
thread.join(timeout=thread_timeout)
```

Reply:
> It looks like you're trying to impose a total time limit on the overall run_dir invocation, but I'm not sure why you'd want to do that.

I'm not sure how long each child should take (i.e is there a use case where one join could take longer than an hour?) Whereas, I don't believe the entire process should last more than a day, which, like you suggested (I'm assuming) is already obsessive. 

Though, I'm fine doing it this way. Fixing.

Reply:
> I'd remove the conditional breaks.

_N.B_, we still need the last condition 
```
if thread.isAlive(): break
```
which checks to see if we timed out. Unless I misunderstood the workflow?

Reply:
My understanding is that `thread.isAlive()` can be used to determine if the `join()` timed out or returned because the thread completed. `join()` would return with `None` regardless, so it wouldn't cause flow control issues. 

We could (and should) use that conditional to log when we encounter a timeout, however. Maybe something like this?

```python
# add name for identifying timeouts
thread = threading.Thread(
  name=run_name,
  target=run_file,
  args=args,
  kwargs=kwargs
)


if thread.isAlive():
  log.error("Timeout executing {}", thread.name)
```

Reply:
Nice touch, thank you.